source/extensions/omni.isaac.orbit/omni/isaac/orbit/devices/gamepad/se3_gamepad.py

```diff
@@ -0,0 +1,203 @@
+        """
+        # store inputs
+        self.pos_sensitivity = pos_sensitivity
+        self.rot_sensitivity = rot_sensitivity
+        self.deadzone = deadzone
+        # acquire omniverse interfaces
+        self._appwindow = omni.appwindow.get_default_app_window()
+        self._input = carb.input.acquire_input_interface()
+        self._gamepad = self._appwindow.get_gamepad(0)
+        self._gamepad_sub = self._input.subscribe_to_gamepad_events(self._gamepad, self._on_gamepad_event)
+        # bindings for gamepad to command
+        self._create_key_bindings()
+        # command buffers
+        self._close_gripper = False
+        self._delta_pos = np.zeros(3)  # (x, y, z)
+        self._delta_rot = np.zeros(3)  # (roll, pitch, yaw)
+        # dictionary for additional callbacks
+        self._additional_callbacks = dict()
+
+    def __str__(self) -> str:
+        """Returns: A string containing the information of joystick."""
+        msg = f"Gamepad Controller for SE(3): {self.__class__.__name__}\n"
+        msg += "\tToggle gripper (open/close): B/A\n"
+        msg += "\tMove arm along x-axis: Left Stick Up/Down\n"
+        msg += "\tMove arm along y-axis: Left Stick Left/Right\n"
+        msg += "\tMove arm along z-axis: Right Stick Up/Down\n"
+        msg += "\tRotate arm along x-axis: D-Pad Left/Right\n"
+        msg += "\tRotate arm along y-axis: D-Pad Up/Down\n"
+        msg += "\tRotate arm along z-axis: Right Stick Left/Right\n"
+        return msg
+
+    """
+    Operations
+    """
+
+    def reset(self):
+        # default flags
+        self._close_gripper = False
+        self._delta_pos = np.zeros([2, 3])  # (pos, neg) (x, y, z)
+        self._delta_rot = np.zeros([2, 3])  # (pos, neg) (roll, pitch, yaw)
+
+    def add_callback(self, key: str, func: Callable):
+        """Add additional functions to bind gamepad.
+
+        A list of available keys are present in the
+        `carb documentation <https://docs.omniverse.nvidia.com/kit/docs/carbonite/latest/docs/python/carb.html?highlight=gamepadeventtype#carb.input.GamepadInput>`_.
+
+        The callback function should not take any arguments.
+
+        Args:
+            key (str): The gamepad button to check against.
+            func (Callable): The function to call when key is pressed.
+        """
+        self._additional_callbacks[key] = func
+
+    def advance(self) -> Tuple[np.ndarray, bool]:
+        """Provides the result from gamepad event state.
+
+        Returns:
+            Tuple[np.ndarray, bool] -- A tuple containing the delta pose command and gripper commands.
+        """
+        delta_rot_sgn = self._delta_rot[0, :] > self._delta_rot[1, :]
+        delta_rot = self._delta_rot.max(axis=0)
+        delta_rot[~delta_rot_sgn] *= -1
+        delta_pos_sgn = self._delta_pos[0, :] > self._delta_pos[1, :]
+        delta_pos = self._delta_pos.max(axis=0)
+        delta_pos[~delta_pos_sgn] *= -1
```

Comment:
Can you add doc-strings here explaining what you're trying to do? Not so obvious otherwise.

Reply:
Thanks a lot for your suggestion. I did so in the new commit.